pkg/appgw/test_fixtures.go

```diff
@@ -164,3 +180,148 @@ func makeIngressTestFixture() v1beta1.Ingress {
 		},
 	}
 }
+
+func makeServicePorts() (v1.ServicePort, v1.ServicePort, v1.ServicePort, v1.ServicePort) {
+	port1 := v1.ServicePort{
+		// The name of this port within the service. This must be a DNS_LABEL.
+		// All ports within a ServiceSpec must have unique names. This maps to
+		// the 'Name' field in EndpointPort objects.
+		// Optional if only one ServicePort is defined on this service.
+		Name: "http",
+
+		// The IP protocol for this port. Supports "TCP", "UDP", and "SCTP".
+		Protocol: v1.ProtocolTCP,
+
+		// The port that will be exposed by this service.
+		Port: 80,
+
+		// Number or name of the port to access on the pods targeted by the service.
+		// Number must be in the range 1 to 65535. Name must be an IANA_SVC_NAME.
+		// If this is a string, it will be looked up as a named port in the
+		// target Pod's container ports. If this is not specified, the value
+		// of the 'port' field is used (an identity map).
+		// This field is ignored for services with clusterIP=None, and should be
+		// omitted or set equal to the 'port' field.
+		TargetPort: intstr.IntOrString{
+			IntVal: 8181,
+		},
+	}
+
+	port2 := v1.ServicePort{
+		Name:     "https",
+		Protocol: v1.ProtocolTCP,
+		Port:     443,
+		TargetPort: intstr.IntOrString{
+			Type:   intstr.String,
+			StrVal: "https-port",
+		},
+	}
+
+	port3 := v1.ServicePort{
+		Name:     "other-tcp-port",
+		Protocol: v1.ProtocolTCP,
+		Port:     554,
+		TargetPort: intstr.IntOrString{
+			Type:   intstr.Int,
+			IntVal: 9554,
+		},
+	}
+
+	port4 := v1.ServicePort{
+		Name:     "other-tcp-port",
+		Protocol: v1.ProtocolUDP,
+		Port:     9123,
+		TargetPort: intstr.IntOrString{
+			Type:   intstr.Int,
+			IntVal: 4566,
+		},
+	}
+
+	return port1, port2, port3, port4
+}
+
+func makePod(serviceName string, ingressNamespace string, containerName string, containerPort int32) *v1.Pod {
+	return &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceName,
+			Namespace: ingressNamespace,
+			Labels: map[string]string{
+				"app": "frontend",
+			},
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  serviceName,
+					Image: "image",
+					Ports: []v1.ContainerPort{
+						{
+							Name:          containerName,
+							ContainerPort: containerPort,
+						},
+					},
+					ReadinessProbe: &v1.Probe{
+						TimeoutSeconds:   5,
+						FailureThreshold: 3,
+						PeriodSeconds:    20,
+						Handler: v1.Handler{
+							HTTPGet: &v1.HTTPGetAction{
+								Host: "bye.com",
+								Path: "/healthz",
+								Port: intstr.IntOrString{
+									Type:   intstr.String,
+									StrVal: containerName,
+								},
+								Scheme: v1.URISchemeHTTP,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func makeService(servicePorts ...v1.ServicePort) *v1.Service {
+	return &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testFixturesServiceName,
+			Namespace: testFixturesNamespace,
+		},
+		Spec: v1.ServiceSpec{
```

Comment:
Can you also add `Selector` so that it can be used to pick up pods ?
```
Selector: map[string]string{
    "app": "frontend",
}
```

Reply:
Added `Selector`!